src/lib/encoder/Decoder.h

```diff
@@ -161,27 +156,27 @@ inline unsigned Decoder::do_decode_coupled(CimbReader& reader, STREAM& ostream)
 }
 
 template <typename MAT, typename STREAM>
-inline unsigned Decoder::decode(const MAT& img, STREAM& ostream, bool should_preprocess, int color_correction)
+inline unsigned Decoder::decode(const MAT& img, STREAM& ostream, unsigned color_mode, bool should_preprocess, int color_correction)
 {
-	CimbReader reader(img, _decoder, should_preprocess, color_correction);
-	return do_decode(reader, ostream);
+	CimbReader reader(img, _decoder, color_mode, should_preprocess, color_correction);
+	return do_decode(reader, ostream, color_mode==0);
```

Comment:
This `color_mode==0` check is what determines where we do the "coupled" decode -- I don't think it *really* makes sense to lock them together that way long term, so this is something that will hopefully be addressed by future changes to the way runtime configuration is done.